File: Python/2.3.1-完全切分/Main.py

from jpype import JClass
def fully_segment(text, dic):
    word_list = []
    for i in range(len(text)):  # 遍历text中的所有位置下标
        for j in range(i + 1, len(text) + 1):  # 遍历[i + 1, len(text)]区间
            word = text[i:j]  # 取出连续区间[i, j]对应的字符串
            if word in dic:  # 如果存在于词典中，则认为是一个词
                word_list.append(word)
    return word_list

File: Python/2.3.1-完全切分/src/Main.java
import com.hankcs.hanlp.dictionary.CoreDictionary;

import java.util.LinkedList;
import java.util.List;
import java.util.TreeMap;

public class Main
{
    public static List<String> segmentFully(String text, TreeMap<String, CoreDictionary.Attribute> dictionary)
    {
        List<String> wordList = new LinkedList<String>();
        for(int i = 0; i < text.length(); ++i)
        {
            for(int j = i + 1; j <= text.length(); ++j)
            {
                String word = text.substring(i, j);
                if(dictionary.containsKey(word))
                    wordList.add(word);
            }
        }
        return wordList;
    }

}
